rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -1095,8 +1093,8 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> gui_settings, std
 	};
 
 	// Events
-	connect(lib_mode_bg, &QButtonGroup::idClicked, on_lib_button_clicked);
-	connect(ui->searchBox, &QLineEdit::textChanged, on_search_box_text_changed);
+	connect(lib_mode_bg, &QButtonGroup::idToggled, [this, on_lib_state_changed](int id){ on_lib_state_changed(ui->searchBox->text(), id); });
```

Comment:
idToggled is more appropriate here I think.

Reply:
A: the function args are wrong.
B: no it is not. with this you'll call state changed on programmatical changes, which we usually want to prevent

Reply:
to clarify:
Qt itself calls click() when the user changes selection (for example by mouse or keyboard).
If you now use toggle, you'll trigger the signal whenever we change the value in code, which might add weird errors in the future. We then would have to add a signal blocker when we don't want the slot being called....

So it's much easier to just call the slot ourselves if necessary after we are finished toggling buttons.

Reply:
Fixed lambda args, thanks. I want to invalidate selection state *only* after switching mode and not just random clicks on the same setting so I think idToggled is better for this..

Reply:
oh hmm I'll think of something better without this signal.

Reply:
if you're worried about selection state during the dialog then you're doing something wrong.
The global library state should only be altered on apply_configs.
Calling the lambda with the current selection after initialization should not have any effect.

Reply:
Fixed.